tests/periph_timer_cli/main.c

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for periph timer API
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+
+#include "shell.h"
+#include "periph/timer.h"
+#include "periph/gpio.h"
+#include "mutex.h"
+
+#define ARG_ERROR       (-1)
+#define CONVERT_ERROR   (-32768)
+#define RESULT_OK       (0)
+#define RESULT_ERROR    (-1)
+#define INVALID_ARGS    puts("Error: Invalid number of arguments");
+
+#define CB_TOGGLE_STR   "cb_toggle"
+#define CB_HIGH_STR     "cb_high"
+#define CB_LOW_STR      "cb_low"
+
+static mutex_t cb_mutex;
+
+static inline int _get_num(const char *str)
+{
+    errno = 0;
+    char *temp;
+    long val = strtol(str, &temp, 0);
+
+    if (temp == str || *temp != '\0' ||
+        ((val == LONG_MIN || val == LONG_MAX) && errno == ERANGE)) {
+        val = CONVERT_ERROR;
+    }
+    return (int)val;
+}
+
+static int _check_param(int argc, char **argv, int c_min, int c_max, char *use)
+{
+    int dev;
+
+    if (argc - 1 < c_min || argc - 1 > c_max) {
+        printf("Usage: %s %s\n", argv[0], use);
+        INVALID_ARGS;
+        return ARG_ERROR;
+    }
+
+    dev = _get_num(argv[1]);
+    if (dev < 0 || dev >= (int)TIMER_NUMOF) {
+        printf("Error: No device, only %d supported\n", (int)TIMER_NUMOF);
+        return ARG_ERROR;
+    }
+    return dev;
+}
+
+static int _print_cmd_result(const char *cmd, bool success, int ret)
+{
+    printf("%s: %s(): [%d]\n", success ? "Success" : "Error", cmd, ret);
+    return success ? RESULT_OK : RESULT_ERROR;
+}
+
+static gpio_t _get_pin(const char *port_str, const char *pin_str)
+{
+    int port = _get_num(port_str);
```

Comment:
error case is not checked here and below, i.e. `CONVER_ERROR`

Reply:
done